# HG changeset patch
# User Colin Macdonald <cbm@m.fsf.org>
# Date 1551728538 28800
#      Mon Mar 04 11:42:18 2019 -0800
# Branch stable
# Node ID 55e087e232fef48f5be0375c240c3ff1457590c8
# Parent  11e7037bed010ff474976cfff4838fbddd5ded48
doc: pkg versions can be more general than "x.y.z" (bug #55798)

* package.txi (The DESCRIPTION File): Document the characters allowed in pkg
versions.  Add reference to the compare_versions function.

diff --git a/doc/interpreter/package.txi b/doc/interpreter/package.txi
--- a/doc/interpreter/package.txi
+++ b/doc/interpreter/package.txi
@@ -353,18 +353,22 @@ License: GPLv3+
 
 The package manager currently recognizes the following keywords
 
 @table @code
 @item Name
 Name of the package.
 
 @item Version
-Version of the package.  A package version must be 3 numbers separated
-by dots.
+Version of the package.  A package version is typically digits separated by
+dots but may also contain @samp{+}, @samp{-}, @samp{~}, and alphanumeric
+characters (in the "C" locale).  For example, @qcode{"2.1.0+"} could indicate
+a development version of a package.
+@c regexp in get_description.m:is_valid_pkg_version_string
+Versions are compared using @ref{XREFcompare_versions,,compare_versions}.
 
 @item Date
 Date of last update.
 
 @item Author
 Original author of the package.
 
 @item Maintainer
